addons/missionTest/functions/fnc_onMissionSave.sqf

```diff
@@ -62,18 +85,28 @@ if (!(_classesNone isEqualTo [])) then {
 };
 
 
-// Verify units have weapons:
+// Verify units have weapons and enough magazines:
 private _checkWeapons = [];
 private _checkMagazines = [];
+private _checkMedical = [];
 {
     private _unit = _x;
-    if ((side _unit) != civilian) then {
+    private _typeOf = toLower typeOf _unit;
+    if ((side _unit) != civilian) then { // Ignore all civilians
+        // Check all units have a primary weapon:
         if ((primaryWeapon _unit) == "") then {
-            if ((typeOf _unit) == "potato_msv_pol") exitWith {};
-            WARNING_1("[%1] has no rifle", typeOf _unit);
-            _checkWeapons pushBackUnique (typeOf _unit);
+            if (_typeOf == "potato_msv_pol") exitWith {};
+            TRACE_1("no rifle",_typeOf);
+            _checkWeapons pushBackUnique format ["%1 has no primaryWeapon", _typeOf];
         };
-        {
+        // Check AT actualy have some kind of AT
+        if ((_typeOf find "soldier_at_f" > 0) || {_typeOf find "soldier_lat_f" > 0} || {_typeOf find "msv_g" > 0} || {_typeOf find "msv_matg" > 0}) then {
```

Comment:
Should these checks be against `-1`? The classnames should all have some level of prefixing but it implies the wrong condition?